R/arith.R

```diff
@@ -60,7 +61,21 @@ Ops.units <- function(e1, e2) {
     units(e2) <- units(e1) # convert before we can compare; errors if unconvertible
   }
 
-  if (mod) {
+  identical_units <-
+    inherits(e1, "units") & inherits(e2, "units") &&
+    identical(units(e1), units(e2))
+
+  if (div & identical_units) {
+    # Special cases for identical units which may not be otherwise divisible by
+    # udunits (see #310)
+    if (.Generic == "%/%") {
+      return(set_units(as.numeric(e1) %/% as.numeric(e2), ""))
```

Comment:
We cannot use `as.numeric` in general, because it breaks compatibility with `quantities` (see the check errors). Please use `drop_units` instead in both arguments instead. Then, we can just call `NextMethod()`, and there is no need for branches.

Regarding "unitless", please use a `1` as we do down below in the "prd" branch. Just a matter of consistency.

Reply:
For going to `NextMethod()`, wouldn't it be preferable to have the output as a unitless number (the output I have) rather than a bare number?  That was what I was going for, but maybe you prefer a bare number (or maybe I'm missing something).

Reply:
`set_units(NextMethod(), 1)`

Reply:
Neat!  I just learned how NextMethod works!  Thanks!